### DISCUSSION TITLE
Interface w/other apps // offline data

### DISCUSSION THREAD

**Discussion by 3535j** [SOTAmat/SOTAcat#57] [Ideas]
Random thoughts:
-Feed of offline data from say an inreach when not connected and presumably turn off the pinging for a connection i.e. other current summits/spots on the air
   -tie to InReach messenger app, txt, cut/paste, etc? 
-wifi CAT tie in to other apps such as iFTx to eliminate cables the same. https://apps.apple.com/us/app/iftx/id6446093115  FT8 doesn't need a computer interface w/this.

**Comment by jeffkowalski** [comment]
Thank you for your feedback!  responses below

_1) Feed of offline data from say an inreach when not connected and presumably turn off the pinging for a connection i.e. other current summits/spots on the air_

This sounds like two different items:
a) find a way to connect to internet via inreach.  I don't have one of those, so I would need more characterization from you about how that device works and presents itself (as a hotspot?).  @brianmathews (AB6D) might know more here.  He has one.
b) disable the refreshing of the SOTA/POTA lists.  Great idea.  I'll create an "issue" here on github to track that feature.

_2) tie to InReach messenger app, txt, cut/paste, etc?_

Again, here we'd need to understand more of what you'd want do...not sure what txt, and cut/paste might mean in the context of SOTAcat.

_3) wifi CAT tie in to other apps such as iFTx to eliminate cables the same. https://apps.apple.com/us/app/iftx/id6446093115 FT8 doesn't need a computer interface w/this._

This looks like a lovely app.  One thing to know about current hardware in the SOTACAT is that it communicates only through the serial port of the radio, not the audio in nor out.  The way it creates FT8 transmissions is through direct modulation:  it strikes a CW tone, then varies the VFO subtly and rapidly to create the message; there is no "audio" involved.  As a corollary to that, we don't get any audio from the radio, so we can't decode any FT8 signals.  Now one could argue that we should create a two-jack SOTAcat - with an extra male plug that inserts in to the audio jack.  Then we'd have to decode the FT8 on the SOTACAT MCU, or send it along to a device connected by wifi or bluetooth.  In either case, there is much exploring left to be done here.  Can you help?

**Reply by 3535j** [reply]
-Refresh cutoff is definitely a separatable issue and probably a battery saver when no connection

2) The txt cut/paste is of the thought of getting an untethered "offline" dataset from an inreach message or other that could be cut and pasted to feed the sota/pota list.  Certainly wouldn't want to be doing a bunch, but one or two might get a summit to summit.  Inreach couldn't be used as a hotspot per se, just a data source to manually feed very sparingly,

3) the iFTx app is certainly driven by audio via the phone speaker or input.  I wasn't fully thinking the radio audio out and into an app either analog or digital.  Getting into the realm of not knowing what I'm talking about here.  The KX3 has an rx/iq port that drives a panadapter.  KX2 doesn't.  Not sure if that's something, but still another jack and audio would be more universal.  Happy to help explore, but definitely am below a white belt hack.

**Comment by jeffkowalski** [comment]
I've logged the first issue as #58 

On item 2), I'm not familiar enough with the inreach to know how cut and paste would work.  What device does one use to "cut" the data?  An inreach app on the phone? From my naive conception, it seems that it would be more work than just manually dialing the VFO on the radio to the frequency of interest.

On item 3), we'll need different hardware, certainly.  Sampling I/Q is similar to sampling audio.  The challenge with the current hardware is both that it lacks that signal input and it may lack the RAM to buffer the signal before decoding.  And then there's the decoding...
A more appropriate device may be the MAX3b (and in-the-works MAX4b).   https://www.ki6syd.com/max-3b-qrp-radio
That radio is itself programmable.  In many ways, the SOTAcat is a workaround for radios that are not, like the KX2/3.  It does CW and digital modes, but not phone modes.

**Comment by 3535j** [comment]
👍 #2 & #3 may be too far reaching and biting off too much hope with the extreme limitations on message size and other HW factors.  I need to dig into the 2-way sotacat setup as is.  I’ve never used.  Beyond that and sidestepping the ham/FT8 purist drama, there’s a point where you have to scratch your head a little when spotting/checking-in, getting data over cell, sat, internet to connect to make a ham radio contact.  :-)

Witnessed the MAX last year and will have to scope that again.  Sometimes not talking to people is fun too 😂
